examples/hubert/dataset/hubert_dataset.py

```diff
@@ -339,61 +380,26 @@ def __call__(self, batch: Tuple[Tensor, Tensor, int]) -> Tuple[Tensor, Tensor, T
                 The Tensor of labels of dimension `[batch, seq]`.
                 The Tensor of audio lengths of dimension `[batch,]`.
         """
-        audio_sizes = [sample[0].shape[1] for sample in batch]
         if self.pad:
-            audio_size = max(audio_sizes)
+            num_frames = max([sample[0].shape[1] for sample in batch])
         else:
-            audio_size = min(audio_sizes)
+            num_frames = min([sample[0].shape[1] for sample in batch])
         waveforms, labels, lengths = [], [], []
         for sample in batch:
             waveform, label, length = sample
+            # The MFCC feature is 10ms per frame, while the HuBERT's transformer output
+            # is 20ms per frame. Downsample the KMeans label if it's generated by MFCC features.
             if self.feature_type == "mfcc":
                 label = label[::2]
-            waveform, label, length = self._collate_audio_label(waveform, label, length, audio_size, self.rand_crop)
+            waveform, label, length = _crop_audio_label(waveform, label, length, num_frames, self.rand_crop)
             waveforms.append(waveform)
             lengths.append(length)
             labels.append(label)
-
-        data = torch.zeros(len(batch), audio_size)
-        for i in range(len(waveforms)):
-            data[i][0 : waveforms[i].shape[1]] = waveforms[i][0]
-        lengths = torch.tensor(lengths)
+        # make sure the shapes are the same if not apply zero-padding
+        if not self.pad:
+            assert all([waveform.shape[0] == waveforms[0].shape[0] for waveform in waveforms])
+            assert all([label.shape[0] == labels[0].shape[0] for label in labels])
```

Comment:
Can you give error message, so that users hitting this error know what to do to fix the error.